exporter/trace_exporter.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"go.opencensus.io/trace"
+
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/census-instrumentation/opencensus-service/spanreceiver"
+	tracetranslator "github.com/census-instrumentation/opencensus-service/translator/trace"
+)
+
+type TraceExporter interface {
+	ExportSpanData(node *commonpb.Node, spandata ...*trace.SpanData) error
+}
+
+type toOCExportersTransformer struct {
+	ocTraceExporters []trace.Exporter
+}
+
+type TraceExporterSpanReceiver interface {
+	TraceExporter
+	spanreceiver.SpanReceiver
+}
+
+// OCExportersToTraceExporterSpanReceiver is a convenience function that transforms
+// traditional OpenCensus trace.Exporter-s into an opencensus-service TraceExporter.
+// The resulting TraceExporter ignores the passed in node. To make use of the node,
```

Comment:
Can you give an example on how exporters (like Stackdriver Trace) can use `Node`?

Reply:
@songy23 great question. I've filed #46 but basically Stackdriver Monitoring should be able to use the NodeInfo when we create a new exporter to match up the "opencensus_task" identifier with the node when we create an exporter per language-PID combination and then pass in a DefaultMonitoringLabel so for example
```go
func (e *stackdriverKeeper) exporterForNodeLocked(node *commonpb.Node) (*stackdriver.Exporter, error) {
	langPIDKey := extractLangPIDKey(node)
	if exp, ok := e.perLangPID[langPIDKey]; ok && exp != nil {
		return exp, nil
	}

	nodeLabels := new(stackdriver.Labels)
	nodeLabels.Set("opencensus_task", langPIDKey, "Opencensus task identifier")
	// Otherwise now create the exporter
	sde, err := stackdriver.NewExporter(stackdriver.Options{
		DefaultMonitoringLabels: nodeLabels,
	})
	if err != nil {
		return nil, err
	}
	// Now memoize it
	e.perLangPID[langPIDKey] = sde
	return sde, nil
}
```

Reply:
The AWS X-Ray exporter can also use an option to identify the service name https://docs.aws.amazon.com/xray/latest/api/API_Service.html which we shall be able to retrieve from the Node. However, we also need the Node to be able to generate the various zPages.